pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -339,23 +339,37 @@ func updateCEPUID(scopedLog *logrus.Entry, e *endpoint.Endpoint, localCEP *ciliu
 	// If the Endpoint already owns the CEP (by holding the matching CEP UID reference) then we don't have to
 	// worry about other ownership checks.
 	//
-	// This will cover cases such as if the NodeIP changes (as with a reboot). In which case we can safely
-	// take ownership and overwrite the CEPs status.
+	// This will cover cases such as if the NodeIP changes (as with a reboot).
+	// In which case we can safely take ownership and overwrite the CEPs
+	// status. However if the cilium endpoints were previously checkpointed
+	// into tmpfs this check will fail and we will rely on the next check to
+	// prevent us from hijacking CEPs.
 	cepUID := e.GetCiliumEndpointUID()
 	if cepUID == localCEP.UID {
 		return nil
 	}
 
-	var nodeIP string
-	if netStatus := localCEP.Status.Networking; netStatus == nil {
-		return fmt.Errorf("endpoint sync cannot take ownership of CEP that has no nodeIP status")
-	} else {
-		nodeIP = netStatus.NodeIP
-	}
-
 	// We do not want to take ownership of CEPs created on other Nodes.
-	if nodeIP != node.GetCiliumEndpointNodeIP() {
-		return fmt.Errorf("endpoint sync cannot take ownership of CEP that is not local (%q)", nodeIP)
+	// However we can't directly compare the CEP node ip with the node, because
+	// the node ip can change, orphaning the CEP. So we retrieve the pod for
+	// the CEP and compare its node IP with that of the node. The kubelet on
+	// this node will update the pod object appropriately, allowing this check
```

Comment:
Just a thought, what are the core assumptions of how the Kubelet works that we'd be basing this on.  Are we confident that won't change in future releases?

Reply:
The main assumption is that when/if the host ip changes, the kubelet will take notice and populate the updated hostIP in the Kubernetes pod object. Since hostIP is part of the api, I feel like if this isn't happening it would be a kubelet bug.

Reply:
the kubelet HostIP value is set at startup and inmutable after that, in order to change the IP, kubelet has to restart ref https://github.com/kubernetes/kubernetes/pull/93898

This affects hostNetwork pods, that inherit their PodIPs from the kubelet HostIPs.

If kubelet restarts with a new IP, all hostNetwork pods will have the new IP in the pod.status.PodIPs and pod.status.HostIP and the non-hostNetwork pods will have the new IP in pod.status.HostIP